beamer/agent/agent.py

```diff
@@ -188,6 +188,9 @@ def start(self) -> None:
             event_monitor.start()
         self._stopped.clear()
 
+    def get_directions(self) -> tuple[TransferDirection, ...]:
+        return tuple(self._event_processors.keys())
```

Comment:
You should omit the `.keys()` part -- iteration over a `dict` is iteration over its keys and is the preferred way to do it.